### PR TITLE
React tag filtering

### PR DESCRIPTION
**Pull request by bradleypickard**
Implements Tag Filtering with sorting for Feed and Map pages.

Sorting by New looks funky when you test it, but the problem lies with the timestamps themselves. 50% of the uploaded recipes have the same timestamp, and the other 50% share a different timestamp. Not sure why this is happening.

Slight bug reopens last opened Infowindow when changing filters on the map, but after re-closing it once it stays gone when further filter changes are made. The similar problem that was occurring when searching by Location no longer happens to me.

### CODE REVIEW

File: src/main/java/com/google/sps/data/FirestoreDB.java [hunk 1]
```diff
@@ -207,13 +208,43 @@ public Boolean getUserProperty(String userId, String recipeId, String mapName, T
...
       List<String> tagIds, SortingMethod sortingMethod, int page) {
     Query recipesQuery = recipesMatchingTags(tagIds, tagIds.iterator());
-    return getRecipeMetadataQuery(recipesQuery, sortingMethod, page);
+    List<RecipeMetadata> results = getRecipeMetadataQuery(recipesQuery, SortingMethod.NONE, page);
+
+    switch (sortingMethod) {
+      case TOP:
+        System.out.println("Sorting by: TOP");
+        Collections.sort(
+            results, Collections.reverseOrder(Comparator.comparingLong(RecipeMetadata::getVotes)));
```
Comment (hari387) [inline src/main/java/com/google/sps/data/FirestoreDB.java:217]: Why is this necessary when `getRecipeMetadataQuery` already sorts and uses the firestore api to sort?

Comment (bradleypickard) [reply src/main/java/com/google/sps/data/FirestoreDB.java:217]: This is where the Firestore concept of a custom index comes into play. If we make a request filtering by the contents of the _tagIds_ field, while sorting on the _votes_ or _timestamp_ field, the request fails and Firestore prompts you to create a custom index. However, a custom index only works to resolve this problem for a single tag and sorting field combo (ie _tagIds.DiabfQ87RYNP6Ehre8Oi_ and _timestamp_). This is insufficient for the flexibility we want for a user searching by any combination of tags
